mongo/options/clientencryptionoptions.go

```diff
@@ -29,6 +35,48 @@ func (c *ClientEncryptionOptions) SetKmsProviders(providers map[string]map[strin
 	return c
 }
 
+// SetTLSConfig specifies tls.Config instances for each KMS provider to use to configure TLS on all connections created
+// to the KMS provider.
+//
+// This should only be used to set custom TLS configurations. By default, the connection will use an empty tls.Config{} with MinVersion set to tls.VersionTLS12.
+func (c *ClientEncryptionOptions) SetTLSConfig(tlsOpts map[string]*tls.Config) *ClientEncryptionOptions {
+	tlsConfigs := make(map[string]*tls.Config)
+	for provider, config := range tlsOpts {
+		// use TLS min version 1.2 to enforce more secure hash algorithms and advanced cipher suites
+		config.MinVersion = tls.VersionTLS12
```

Comment:
I suggest only overriding if `MinVersion` is unset. A user may request a `MinVersion` of higher than `VersionTLS12`. 

```suggestion
		if config.MinVersion == 0 {
		    config.MinVersion = tls.VersionTLS12
	        }
```